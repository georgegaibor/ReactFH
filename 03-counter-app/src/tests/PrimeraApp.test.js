//import { render } from "@testing-library/react";
import React from 'react';
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme'


describe('PrimeraApp Tests', () => { 
    /*
    USING JEST EXTENSIONS
    test('Should display "Hola"', () => { 

        const greeting = "Hola";
        const {getByText} = render(<PrimeraApp saludo = {greeting}/>);

        expect(getByText(greeting)).toBeInTheDocument();
    });
    */
    
    test('Should display <PrimeraApp/> correctly', () => { 
        const greeting = 'Hello';
        const wrapper = shallow(<PrimeraApp saludo = {greeting}/>);

        expect(wrapper).toMatchSnapshot();

    })

    test('Should display subtitle specified in props', () => { 
        const greeting = 'Hello';
        const subtitle = 'SUBTITLES';

        const wrapper = shallow(<PrimeraApp 
            saludo = {greeting}
            subtitulo = {subtitle} 
        />);

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitle);

    })


})