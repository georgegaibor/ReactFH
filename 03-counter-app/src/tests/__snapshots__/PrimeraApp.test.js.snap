// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp Tests Should display <PrimeraApp/> correctly 1`] = `
<Fragment>
  <h1>
    Hello
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
